core/src/test/java/org/apache/calcite/test/SqlValidatorTest.java

```diff
@@ -11236,6 +11236,35 @@ private void checkCustomColumnResolving(String table) {
     assertThat(resultType.toString(), is("INTEGER"));
   }
 
+  @Test public void testGroupByAlias() {
+    // test group by alias, when the underlying column in the expression is the same as the alias
+    sql("select DEPTNO+1 as DEPTNO from emp group by DEPTNO+1")
+        .withConformance(SqlConformanceEnum.LENIENT)
+        .ok();
```

Comment:
As discussed from JIRA:
1. All projection fields must either be in the groupping columns or in the agg functions
2. Property of `isGroupByAlias` in LENIENT conformance is true.

In my understanding, this Sql should not pass the validation.

Reply:
DEPTNO+1 is a projection field and in group by columns

Reply:
you means case is this?

psql> select a + b a from test group by a;
ERROR:  column "test.b" must appear in the GROUP BY clause or be used in an aggregate function

Reply:
when isGroupByAlias=true, it means you will group by alias, not the underlying column

Reply:
yes, it is.  when isGroupByAlias=true, it means you will group by alias, not the underlying column.

after the fix, such SQL will not has SqlValidatorException，because it will think DEPTNO on group by as a basic column, not a alias
```sql select DEPTNO+1 as DEPTNO from emp group by DEPTNO+1 ```

before the fix, it think DEPTNO on group by is a alias

thank your review

Reply:
In LENIENT, Calcite's behavior should be very much like PostgreSQL. PostgreSQL is group by underlying column.

so let isGroupByAlias in LENIENT conformance is false can make it right.

thank your review

Reply:
When I run the following two SQLs on PG, I found that PG will not throw Exception, so I think that PG does not strictly stipulate whether it is group by alias or group by underlying column . I think it is the priority group byunderlying column, if there is no underlying column, then group by alias.

``` select empno as d from emp group by d ```

``` select empno as d from emp group by  empno```